From 34a5929b285b7ee59867c288dc6717f2db4b5d4e Mon Sep 17 00:00:00 2001
From: Nicolas Hug <nicolashug@fb.com>
Date: Tue, 6 Apr 2021 08:30:45 +0100
Subject: [PATCH] avoid check if pil version is < 8.2 as the reference image
 would be different

Due to inability of conda-build to apply binary patch,
invert condition from pytorch/vision#3631

Adapted-By: H. Vetinari <h.vetinari@gmx.com>
---
 test/test_utils.py | 11 ++++++++---
 1 file changed, 8 insertions(+), 3 deletions(-)

diff --git a/test/test_utils.py b/test/test_utils.py
index 662ad2a0..31fb4212 100644
--- a/test/test_utils.py
+++ b/test/test_utils.py
@@ -7,7 +7,10 @@ import torchvision.utils as utils
 import unittest
 from io import BytesIO
 import torchvision.transforms.functional as F
-from PIL import Image
+from PIL import Image, __version__ as PILLOW_VERSION
+
+
+PILLOW_VERSION = tuple(int(x) for x in PILLOW_VERSION.split('.'))
 
 
 class Tester(unittest.TestCase):
@@ -93,8 +96,10 @@ class Tester(unittest.TestCase):
             res = Image.fromarray(result.permute(1, 2, 0).contiguous().numpy())
             res.save(path)
 
-        expected = torch.as_tensor(np.array(Image.open(path))).permute(2, 0, 1)
-        self.assertTrue(torch.equal(result, expected))
+        if PILLOW_VERSION < (8, 2):
+            # The reference image is only valid for older PIL versions
+            expected = torch.as_tensor(np.array(Image.open(path))).permute(2, 0, 1)
+            self.assertTrue(torch.equal(result, expected))
 
 
 if __name__ == '__main__':
-- 
2.31.1.windows.1

